core/src/main/java/io/github/tcdl/ChannelManager.java

```diff
@@ -53,10 +53,22 @@ public Producer findOrCreateProducer(final String topic) {
         return producer;
     }
 
-    public Consumer findOrCreateConsumer(final String topic) {
+    /**
+     * Start consuming messages on specified topic with handler .
+     * Calls to subscribe() and unsubscribe() have to be properly synchronized by client code not to lose messages.
```

Comment:
I don't completely understand why do we need synchronization between subscribe and unsubscribe.